lib/net/ldap/connection.rb

```diff
@@ -36,7 +36,7 @@ def prepare_socket(server)
     encryption = server[:encryption]
 
     @conn = socket
-    setup_encryption encryption if encryption
+    setup_encryption({ verify_host: server[:connected_host] }.merge(encryption)) if encryption
```

Comment:
No, this is passing the hostname through as `verify_host`, *unless* the user has set `verify_host` to something else.

It's a bit convoluted, but it's on by default, and enabled the user to opt-out.

I would agree having a separate user targeted flag would make sense, but you don't seem to be understanding how this code is working.